src/borg/testsuite/platform.py

```diff
@@ -44,6 +46,27 @@ def user_exists(username):
         return False
 
 
+def are_acls_working():
+    global _acls_working
+    if _acls_working is None:
+        _acls_working = False
+        try:
+            tmpfile = generate_tempfile_path()
+            open(tmpfile, 'w').close()
+            access = b'user::rw-\ngroup::r--\nmask::rw-\nother::---\nuser:root:rw-:9999\ngroup:root:rw-:9999\n'
+            acl = {'acl_access': access}
+            acl_set(tmpfile, acl)
+            read_acl = {}
+            acl_get(tmpfile, read_acl, os.stat(tmpfile))
+            read_acl_access = read_acl.get('acl_access', None)
+            if read_acl_access and b'user::rw-' in read_acl_access:
+                _acls_working = True
+            os.unlink(tmpfile)
+        except PermissionError:
+            pass
```

Comment:
maybe move tmpfile creation before the "try".
move the tmpfile deletion to "finally".